targets/TARGET_RENESAS/TARGET_RZ_A1XX/pwmout_api.c

```diff
@@ -227,12 +229,35 @@ void pwmout_write(pwmout_t* obj, float value) {
             // Do Nothing
         }
         wk_cycle = (uint32_t)*p_mtu2_ctl->period1;
+        wk_pulse = (uint32_t)*p_mtu2_ctl->pulse1;
+        if ((obj->ch == 4) || (obj->ch == 3)) {
+            tmp_tstr_st = (1 << (obj->ch + 3));
+        } else {
+            tmp_tstr_st = (1 << obj->ch);
+        }
 
         // set channel match to percentage
         if (value == 1.0f) {
-            *p_mtu2_ctl->pulse1 = (uint16_t)(wk_cycle - 1);
+            if (wk_pulse != wk_cycle) {
+                MTU2TSTR &= ~tmp_tstr_st;
+                *p_mtu2_ctl->tior = 0x66;
+            }
+        } else if (value == 0.0f) {
+            if (wk_pulse != 0) {
+                MTU2TSTR &= ~tmp_tstr_st;
+                *p_mtu2_ctl->tior = 0x11;
+            }
         } else {
-            *p_mtu2_ctl->pulse1 = (uint16_t)((float)wk_cycle * value);
+            if ((wk_pulse == wk_cycle) || (wk_pulse == 0)) {
+                MTU2TSTR &= ~tmp_tstr_st;
+                *p_mtu2_ctl->tior = 0x65;
+            }
+        }
+        *p_mtu2_ctl->pulse1 = (uint16_t)((float)wk_cycle * value);
+
+        // Counter Restart
+        if ((MTU2TSTR & tmp_tstr_st) == 0) {
+            MTU2TSTR |= tmp_tstr_st;
         }
```

Comment:
By the TIOR register, the output is fixed to Low when the duty is 0%, and fixed to Hi when 100%.
To change the TIOR register, the counter is temporarily stopped.